src/admin.go

```diff
@@ -230,7 +388,252 @@ func UpdateRoute(options Options, dist cloudfront.DistributionSummary) error {
+						//this cert was created by stout, take it!
+						return certificateARN, nil
+					}
+				}
+			}
+		}
+		return "", nil
+	}
+
+	//pick all suitable certificates
+	for _, certificate := range certificatesResponse.CertificateSummaryList {
+		//request certificate details
+		domainName := *certificate.DomainName
+		certificateARN := *certificate.CertificateArn
+
+		//request certificate detail for all
+		certificateDetail, err := acmService.DescribeCertificate(&acm.DescribeCertificateInput{
+			CertificateArn: amazonaws.String(certificateARN),
+		})
+
+		if err != nil {
+			return "", err
+		}
+
+		//domain name exactly matches the cert name
+		if domainName == options.Bucket {
+			return certificateARN, nil
+		}
+
+		for _, certSAN := range certificateDetail.Certificate.SubjectAlternativeNames {
+			if *certSAN == options.Bucket {
+				return certificateARN, nil
+			}
+		}
+
+		//domain name falls under a wildcard domain
+		wildcardDomainTLDPlusOne, err := publicsuffix.EffectiveTLDPlusOne(options.Bucket)
+		if err != nil {
+			return "", err
+		}
+		wildCardOfGivenDomain := strings.Join([]string{"*", wildcardDomainTLDPlusOne}, ".")
+
+		if wildCardOfGivenDomain == domainName {
+			return certificateARN, nil
+		}
+
+	}
+	return "", nil
+}
+
+/*
+* Check if the certificate is issued(confirmed)
+ */
+func validateCertificate(acmService *acm.ACM, certificateARN string) error {
+	certificateDetail, err := acmService.DescribeCertificate(&acm.DescribeCertificateInput{
+		CertificateArn: amazonaws.String(certificateARN),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if *certificateDetail.Certificate.Status != "ISSUED" {
+		//the certificate was not issued yet prompt to check their email
+		validationEmails := make([]string, 0)
+		for _, validationOption := range certificateDetail.Certificate.DomainValidationOptions {
+			for _, validationEmail := range validationOption.ValidationEmails {
+
+				validationEmails = append(validationEmails, *validationEmail)
+			}
+		}
+		vEmails := strings.Join(validationEmails, "\n\t- ")
+		return fmt.Errorf("Certificate not issued yet. Please check one of the following emails or use --no-ssl:\n\t- %s", vEmails)
+	} else {
+		return nil
+	}
+}
+
+/*
+* request a new certificate from ACM
+ */
+func requestCertificate(options Options, acmService *acm.ACM) ([]string, error) {
+	certificateReqResponse, err := acmService.RequestCertificate(&acm.RequestCertificateInput{
+		DomainName: amazonaws.String(options.Bucket),
+		DomainValidationOptions: []*acm.DomainValidationOption{
+			{
+				DomainName:       amazonaws.String(options.Bucket),
+				ValidationDomain: amazonaws.String(options.Bucket),
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = acmService.AddTagsToCertificate(&acm.AddTagsToCertificateInput{
+		CertificateArn: amazonaws.String(*certificateReqResponse.CertificateArn),
+		Tags: []*acm.Tag{
+			{
+				Key:   amazonaws.String("stout"),
```

Comment:
`created-by-stout: true` or `created-by: stout` probably is easier for someone to understand externally.